components/brave_wallet/common/eth_address.cc

```diff
@@ -121,21 +111,19 @@ std::optional<std::string> EthAddress::ToEip1191ChecksumAddress(
 
 std::string EthAddress::ToChecksumAddress(uint256_t eip1191_chaincode) const {
   std::string result = "0x";
-  std::string input;
+  std::string prefix;
 
   if (eip1191_chaincode == static_cast<uint256_t>(30) ||
       eip1191_chaincode == static_cast<uint256_t>(31)) {
     // TODO(jocelyn): We will need to revise this if there are supported chains
     // with ID larger than uint64_t.
-    input +=
+    prefix =
         base::NumberToString(static_cast<uint64_t>(eip1191_chaincode)) + "0x";
   }
 
-  input += std::string(ToHex().data() + 2);
-
-  const std::string hash_str(KeccakHash(input).data() + 2);
-  const std::string address_str =
-      base::ToLowerASCII(base::HexEncode(bytes_.data(), bytes_.size()));
+  const std::string address_str = HexEncodeLower(bytes_.data(), bytes_.size());
```

Comment:
I think the issue is that these small functions are not always guaranteed to have the very semantics we think they have, so the next person tends to have to check what is in them, to be sure. They also create more context to understand code you never saw. This is the clash we get between code that is shorter to write, and code that is easier to read.

There's also the fact that they narrow the interface of what they are calling, for instance, these are the overloads for `HexEncode`.

```cxx
BASE_EXPORT std::string HexEncode(base::span<const uint8_t> bytes);
BASE_EXPORT std::string HexEncode(std::string_view chars);
BASE_EXPORT std::string HexEncode(const void* bytes, size_t size);
```

Reply:
> I would add that there is also a clash between who reads this code on a daily basis and who reads it occasionally. Things being equal, it seems reasonable to shift balance towards the convenience of the first groups of readers.

That's correct but the correct answer is the opposite. The entire codebase should aim to be less context specific so all readers don't feel like they are in a special corner of the codebase with specific contextual practices.

Reply:
```I would add that there is also a clash between who reads this code on a daily basis and who reads it occasionally. Things being equal, it seems reasonable to shift balance towards the convenience of the first groups of reader```

I couldn't disagree with this more strongly. The code should always favor readability for people who are not working on it on a daily basis. In particular in such a large codebase we can't have silos with their own utilities and patterns that people who work on other parts of the code have to learn and understand. Of course you will have domain specific code that people have to learn, but you should strive to make it easier to understand for people outside your team. There are many examples of this principle in chromium, but here is one in reference to naming https://google.github.io/styleguide/cppguide.html#General_Naming_Rules

Reply:
> I couldn't disagree with this more strongly.

You're disagreeing with something I didn't write. In fact I mostly agree with the nature of "silos" argument but it isn't what is context of this debate about.

`Things being equal` means `HexEncodeLower` and `base::ToLowerASCII(base::HexEncode(v))` have same level of readability for *everyone* (that's subjective of course and let's debate on this only if there are strong arguments the first one is unreadable), so `shift balance towards the convenience` means I can pick one of them for the code wallet devs will have to manage.

> you should strive to make it easier to understand for people outside your team

No, I shouldn't in general. That doesn't mean I would intentionally write code which could be understood only by  wallet people. But I don't see it worth costs to be it clear for *everyone*(that's not measurable and achievable). For example it doesn't make sense for every appearance of word `Hardened` explain what it means https://github.com/brave/brave-core/pull/22870#discussion_r1575225249 

`can't have silos with their own utilities and patterns` there is a lot of `something_utils.cc` throughout the code base. 
`std::string GetPlaceholderToken(std::string_view value);` Is stuff like this allowed?

Reply:
Since this thread is being revived I can add a few things, but I'm concerned we may not be getting anywhere, so let me try to address things to the best of my abilities.

> You're disagreeing with something I didn't write

I don't think that's true, based on what has already been said, and what has been added now.

> Things being equal means HexEncodeLower and base::ToLowerASCII(base::HexEncode(v)) have same level of readability for everyone

That's obviously not the case, as I had in fact to check this function's definition, and I was surprised to learn that it was just `base::ToLowerASCII(base::HexEncode(v))`. In the process, I also realised we were not using the safe variants of `base::HexEncode` due to how wrapping things like this hides the actual semantics of the arguments involved.

> For example it doesn't make sense for every appearance of word Hardened explain what it means https://github.com/brave/brave-core/pull/22870#discussion_r1575225249

That's a good example of something being brought up in review, and then an informed determination being made that the special context was appropriate. In this thread/review we have the opposite, where the information has been weighed in and it becomes evident we are actually making code more obscure, and advice is being ignored.

> No, I shouldn't in general.

Actually, you should. We all should. To quote from the Google C++ Guidelines:

> **Optimize for the reader, not the writer** Our codebase (and most individual components submitted to it) is expected to continue for quite some time. As a result, more time will be spent reading most of our code than writing it. We explicitly choose to optimize for the experience of our average software engineer reading, maintaining, and debugging code in our codebase rather than ease when writing said code. "Leave a trace for the reader" is a particularly common sub-point of this principle: When something surprising or unusual is happening in a snippet of code (for example, transfer of pointer ownership), leaving textual hints for the reader at the point of use is valuable (std::unique_ptr demonstrates the ownership transfer unambiguously at the call site).

There's no great subjective question here. The question is simple: Is `HexEncodeLower` obscuring `base::ToLowerASCII(base::HexEncode(v))`? Yes, objectively yes, because that's how I found out we were still using the unsafe variant of `HexEncode`. Would someone in wallets struggle to understand `base::ToLowerASCII(base::HexEncode())`? No. Would someone outside wallets struggle to understand `HexEncodeLower` at first sight? Yes.

So in summary, **we've made code harder to read to the person unfamiliar with wallet, while the alternative, i.e. `base::ToLowerASCII(base::HexEncode(v))`,  wouldn't have made the code any less readable for you or for anybody else.**

> there is a lot of something_utils.cc throughout the code base. std::string GetPlaceholderToken(std::string_view value); Is stuff like this allowed?

In general utils are frowned upon in chromium. There has been attempts to get a namespace `util` in base for instance, and that was eventually removed. Google bans `utils` internally. Why? Because utils can be a large source of code obfuscation.

But back to your original question: Are utils allowed? A util has to arguably make a case that code is now easier to read with it. For the example, you mentioned `GetPlaceholderToken`. Here's the definition for it:

```cxx
std::string GetPlaceholderToken(std::string_view value) {
  return base::StrCat({"${", value, "}"});
}
```

So what we have here? We have a util function, that is very contextual to this particular part of chromium. It hides away this ugly string concatenation. It has this concept of a token being brackets. This function is most certainly improving readability in the component it belongs, as opposed to have that `StrCat` repeated.

Notice though that with `HexEncodeLower` we have none of that. It merely wraps two functions everyone knows behind a new name most of us don't without adding any particular gain to anything wallet specific.

> so shift balance towards the convenience means I can pick one of them for the code wallet devs will have to manage

I have in the past few months submitted a significant amount of changes to wallet, including bug fixes, overall code health changes, improvements, etc. Does that make me a wallet maintainer? We can't have this distinction when the question is code readability.

In my previous comment I said: "This is the clash we get between code that is shorter to write, and code that is easier to read.". You have replied to that with "I would add that there is also a clash between who reads this code on a daily basis and who reads it occasionally.". My statement about short code is very much different from your answer, and it does not make any assertions about how much a say the reader has on complexity. Why? Because it shouldn't matter. Keeping complexity lower for all can in many cases saves time and resources.

On another note, I think things like this are too small, and in general if positive review feedback has been provided so far, which has improved a PR, what would be the point of resisting so much on this? I did in fact reach out during code review on slack about this particular thread. I have not received any reply on this on slack, and now things are still being debated here.